docs/integrations/consul/k8s.md

```diff
@@ -0,0 +1,314 @@
+- In a production environment, or anywhere you wish to use Infrastructure as Code and source control your Helm configurations, you can set up your credentials following this [guide](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/deployment-guide/credentials.md).
+:::
+
+5. Verify the ngrok Ingress Controller is installed and all its pods are healthy
+  ```bash
+  kubectl get pods -l 'app.kubernetes.io/name=kubernetes-ingress-controller' -n consul
+  NAME                                                READY   STATUS    RESTARTS      AGE
+  ngrok-ingress-controller-manager-5b796c88f7-k7v6z   2/2     Running   1 (64s ago)   67s
+  ```
+
+
+## **Step 3**: Install a Sample Application {#install-a-sample-application}
+
+Now let's install a sample application to try out our service mesh and Ingress Controller combination. We'll use the [HashiCups Demo Application](https://github.com/hashicorp-demoapp) HashiCorp uses for demos and guides such as in their [Getting Started with Consul Service Mesh for Kubernetes](https://developer.hashicorp.com/consul/tutorials/kubernetes-features/service-mesh-deploy) guide. This application is a simple e-commerce application that allows users to order coffee cups.
+The application has a frontend and public API services that are also backed by a private API and database. These communicate with each other through the Consul service mesh. This comes with nginx installed as a proxy for the frontend and Public API services. We'll replace this with ngrok to provide public access and other features.
+
+:::tip For this demo, everything will be installed in the `consul` namespace.
+
+The ngrok Ingress Controller can send traffic to services across different namespaces, but [Consul Service Intentions](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions) across namespaces [require an enterprise account](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions#sources-namespace). For now, we'll keep everything in the same namespace.
+:::
+
+1. Clone the [HashiCorp Learning Consul repo](https://github.com/hashicorp/learn-consul-kubernetes). This has multiple great example applications for learning about Consul and Kubernetes. We'll use the HashiCups application for this guide.
+  ```bash
+  git clone https://github.com/hashicorp/learn-consul-kubernetes
+  ```
+1. Install the HashiCups sample app in the `consul` namespace. This app consists of multiple Services and Deployments that make a tiered application. We'll install all of them from this folder and then modify things from there.
+  ```bash
+  kubectl apply -f learn-consul-kubernetes/service-mesh/deploy/hashicups -n consul
+  ```
+
+1. Remove the existing [Service Intentions](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions) for the public-api and frontend services and add our own.
```

Comment:
should `public-api` and `frontend` be surrounded with backticks?

Reply:
I'm not sure if they need to be, but i can add them. They are the actual names of the k8s services and so do map to a resource name of sorts, so it probably makes sense

Reply:
ahh, i do see i added them on some lines, so i'll add them on every normal piece of text to make them consistent